cache/cache.go

```diff
@@ -23,10 +23,16 @@ import (
 type Cache struct {
 	root               string
 	httpClient         *http.Client
-	gh                 *github.Client
 	fastFailHTTPClient *http.Client
+	strategies         []DownloadStrategy
 }
 
+// DownloadStrategy defines a strategy for downloading URLs.
+//
+// Typically useful for packages behind authenticated endpoints, etc.
+// A default download strategy will always be the first strategy attempted.
```

Comment:
> A default download strategy will always be the first strategy attempted.

This feels a bit confusing for me, feels like this should be documented somewhere else? Maybe better without this line?